File: brn.rb
#!/usr/bin/ruby
require 'pathname'

bd = Pathname.new(Dir.pwd)
bd.children.each do |drct|
  rest, prnt = drct.split
  drct.children.each do |f|
    nname = prnt.to_s + ' - ' + f.basename.to_s
    f.rename(nname)
  end
end

File: brn.py
#!/usr/bin/python
import os

for drct in os.listdir('.'):
    for f in os.listdir(drct):
        nname = drct + ' - ' + f
        fnp = os.path.realpath(nname)
        fop = os.path.realpath(drct + '/' + f)
        os.rename(fop, fnp)
